### PR TITLE
Create python-app.yml

### PR DESCRIPTION
**Pull request by TheoMF**

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (nim65s) [inline .github/workflows/python-app.yml:30]:
```suggestion
    - name: Install gepetuto and its dependencies
      run: pipx install -e .
```

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Test with pytest
+      run: |
+        pytest
```
Comment (nim65s) [inline .github/workflows/python-app.yml:39]:
```suggestion
    - name: check with gepetuto
      run: cd tests && gepetuto -vva
```

File: .github/workflows/python-app.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python application
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```
Comment (nim65s) [inline .github/workflows/python-app.yml:10]:
```suggestion
on: [push, pull_request]
```

File: .github/workflows/python-app.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python application
```
Comment (nim65s) [inline .github/workflows/python-app.yml:4]:
```suggestion
name: check gepetuto
```

**Review by nim65s — CHANGES_REQUESTED** [review]
It looks like you forgot to adapt the template to our needs. You can commit my suggestions in the github web interface, or push other ones, as you prefer.

**Comment by TheoMF** [comment]
> It looks like you forgot to adapt the template to our needs. You can commit my suggestions in the github web interface, or push other ones, as you prefer.

that's the first time I'm doing it, I made the commit on github but I didn't had it in local so I couldn't modify it properly at the time

**Review by nim65s — APPROVED** [review]

**Pull request merged**